components/campaign-hero/Component.jsx

```diff
@@ -0,0 +1,243 @@
+ * @param {object} bkgConfig
+ * Background configuration - could be image or looping video from Vimeo
+ *
+ * @param {object} textConfig
+ * Text configuration - title and intro text for the hero
+ *
+ * @param {string} youtubeId
+ * YouTube ID of the video that will play in a modal
+ *
+ * @param {object} quoteConfig
+ * Quote configuration - fields for the optional quote at the bottom of the hero
+ *
+ * @param {object} bkgImageData
+ * Data from the background image - src and alt text
+ *
+ * @param {object} quoteImageData
+ * Data from the quote image - src and alt text
+ *
+ * @param {string} quoteInternalLinkUrl
+ * The internal link URL for the quotee name if using Matrix asset link
+ *
+ * @returns {JSX.Element}
+ * @constructor
+ */
+
+export default function CampaignHero({
+  bkgConfig,
+  textConfig,
+  youtubeId,
+  quoteConfig,
+  bkgImageData,
+  quoteImageData,
+  quoteInternalLinkUrl,
+}) {
+  // Conditionals for code readability
+  const isBgVideo = bkgConfig?.type === "Video" && bkgConfig?.bkgVideo;
+  const hasQuote = quoteConfig?.include && quoteConfig?.quote;
+
+  /**
+   * The condition to display a pulled left intro.
+   * 1. Quote is not included
+   * 2. Background is an image
+   * 3. No YouTube video to play in a modal
+   */
+  const isIntroPulledLeft =
+    !quoteConfig.include && bkgConfig.type === "Image" && !youtubeId;
+
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isUserPaused, setIsUserPaused] = useState(false); // Track if the video was paused by the user
+  const [isPlaying, setIsPlaying] = useState(false);
+  const iframeRef = useRef(null);
+
+  const handleClick = () => {
+    setIsModalOpen(true);
+  };
+
+  const handleCloseModal = () => {
+    setIsModalOpen(false);
+  };
+
+  // Toggles play/pause when button is clicked
+  const togglePlayPause = () => {
+    setIsUserPaused((prev) => !prev); // Track manual pause state
+    setIsPlaying((prev) => !prev); // Toggle play/pause state
+  };
+
+  // Send postMessage to Vimeo iframe to play/pause video
+  useEffect(() => {
+    const iframe = iframeRef.current;
+    if (iframe && iframe.contentWindow) {
+      iframe.contentWindow.postMessage(
+        JSON.stringify({ method: isPlaying ? "play" : "pause" }),
+        "*"
+      );
+    }
+  }, [isPlaying]);
+
+  // Pause video when it's out of viewport using IntersectionObserver
+  useEffect(() => {
+    const observer = new IntersectionObserver(
+      (entries) => {
+        const [entry] = entries;
+
+        // Only set play/pause based on IntersectionObserver if video is not manually paused
+        if (!isUserPaused) {
+          setIsPlaying(entry.isIntersecting);
+        }
+      },
+      { threshold: 0 } // Adjust threshold as needed
+    );
+
+    const iframe = iframeRef.current;
+    if (iframe) observer.observe(iframe);
+
+    return () => {
+      if (iframe) observer.unobserve(iframe);
+    };
+  }, [isUserPaused]); // Re-run the effect if isUserPaused state changes
```

Comment:
I originally took some code from the Single Image Video component to handle the vimeo play pause, but I rewrote it here using React `useRef` to reference the iframe, and intersection observer which seems to make it more reliable (and also using more of the React functionality).